services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -520,6 +520,8 @@ public VeniceHelixAdmin(
         .add(new SystemSchemaInitializationRoutine(AvroProtocolDefinition.PARTITION_STATE, multiClusterConfigs, this));
     initRoutines.add(
         new SystemSchemaInitializationRoutine(AvroProtocolDefinition.STORE_VERSION_STATE, multiClusterConfigs, this));
+    initRoutines
```

Comment:
I'm not sure this is the right way. This is for schema system stores

Reply:
These are the concerns with this:
1. This change is only executed on child controllers. We need push job details initialization to be done in parent controllers
    1. The current push job details initialization is done in [VeniceParentHelixAdmin](https://github.com/linkedin/venice/blob/main/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java#L513). Can you add schema registration logic there?
    2. The current store creation logic is also incorrect in that function. This PR should address that too.
3. Can you enable some multi-colo test to produce push job details? The feature is [disabled in tests currently](https://github.com/linkedin/venice/blob/main/internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestWriteUtils.java#L912) and hence, we never caught the issue earlier (at least in the creation logic)

Reply:
Doesn't VeniceParentHelixAdmin contains a VeniceHelixAdmin instance, so that it would also be handled automatically? If you look at the integration test part, `getAllValueSchema` is invoked on the parent controller client, so that means parent controller should work in this case?

Reply:
I also tried enable PUSH_JOB_STATUS_UPLOAD_ENABLE with this change, and it seems to pass (please check internal CI result for  #7815).

Reply:
If we find that this works fine, then the implementation inside `VeniceParentHelixAdmin` needs to be cleaned up. We should not have competing logics in the hopes that one will clean up the other's mess. This would just lead to race conditions down the line

Reply:
@nisargthakkar, sorry for the long delay on this thread. Getting back to your last feedback, which part in `VeniceParentHelixAdmin` do you refer to as the competing logic for race conditions? Could you give a pointer please, thanks!

Reply:
Hi Lei. This is the competing implementation in `VeniceParentHelixAdmin`: https://github.com/linkedin/venice/blob/main/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java#L535

It has a TODO to remove this and move to init routines, but we should not have both trying to create and update the store.

Can you take a look at what this one does and separate the concerns for the new logic and the old one?

This is what I had on the ticket. Can you take a look at these too?
> It looks like our SchemaInitializationRoutine doesn't update schemas for non-schema system stores: https://github.com/linkedin/venice/blob/main/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java#L628

> (Please verify my understanding as it seems sketchy to me and makes me wonder how this ever worked in the past. Maybe we never evolved schemas after introducing the init routines)

> Update: 
> It looks like that initial store creation logic is incorrect too. The current push job details value schema will be registered as v1 schema in tests: 
[https://github.com/linkedin/venice/blob/4917d41a0033ecba8167e62576f4a54d96dba6ad/s[…]java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java](https://github.com/linkedin/venice/blob/4917d41a0033ecba8167e62576f4a54d96dba6ad/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java#L633)